index_configuration_tool/tests/test_logstash_parser.py

```diff
@@ -0,0 +1,76 @@
+import pickle
+import unittest
+from os.path import dirname
+
+import lark.exceptions
+from jsondiff import diff
+
+from logstash_conf_parser import logstash_parser, parse
+
+# Constants
+LOGSTASH_TEST_INPUT_FILE = dirname(__file__) + "/logstash_test_input.conf"
+LOGSTASH_EXPECTED_OUTPUT_FILE = dirname(__file__) + "/expected_parse_output.pickle"
+
+
+class TestLogstashParser(unittest.TestCase):
+    # Run before each test
+    def setUp(self) -> None:
+        with open(LOGSTASH_EXPECTED_OUTPUT_FILE, "rb") as f:
+            # The root DS is a dict, with input type as key.
+            # The value of each key is an array of inputs.
+            # Each input is a tuple of plugin name and data,
+            # where the data is a dict of key-value pairs.
+            self.test_data = pickle.load(f)
+
+    # Test input json should match loaded pickle data
+    def test_parser_happy_case(self):
+        actual = parse(LOGSTASH_TEST_INPUT_FILE)
+        test_diff = diff(self.test_data, actual)
+        # Validate that diff is empty
+        self.assertFalse(test_diff)
+
+    def test_bad_configs(self):
+        with self.assertRaises(lark.exceptions.UnexpectedToken):
```

Comment:
If one of these raises an `UnexpectedToken` and the rest don't, this test would pass, right?

And currently, if the first raises an exception, do the rest actually execute? I would expect the code to stop running at that point.

Regardless, I think these should be separated to be in their own `with assertRaises` context, unless I'm misunderstanding something about how it works.

Reply:
Good point. I've changed this (and a similar test case in `test_main.py` to use a loop now.